addons/mail/models/mail_channel.py

```diff
@@ -1133,3 +1133,21 @@ def _execute_command_who(self, **kwargs):
             msg = _("Users in this channel: %(members)s %(dots)s and you.", members=", ".join(members), dots=dots)
 
         self._send_transient_message(partner, msg)
+
+    @api.model
+    def _mark_seen_first_msg(self):
+        channel = self.env.ref('mail.channel_all_employees', raise_if_not_found=False)
+        message = self.env.ref('mail.module_install_notification', raise_if_not_found=False)
+        # _set_last_seen_message method takes partner_id of SUPERUSER because this method calls once during installation
+        # But we required admin users to mark first message as read
+        for user in self.env['res.users'].search([('groups_id', 'in', self.env.ref('base.group_erp_manager').ids)]):
+            if channel and message:
+                channel_partner = self.env['mail.channel.partner'].search([
+                    ('channel_id', '=', channel.id),
+                    ('partner_id', '=', user.partner_id.id),
+                    ('seen_message_id', '=', False),
+                ])
+                channel_partner.write({
+                    'fetched_message_id': message.id,
+                    'seen_message_id': message.id,
+                })
```

Comment:
You can mark it as read for all existing users at that time, not just the admin: so remove `partner_id` and `seen_message_id` from the domain and remove the loop, and just write on all channel_partners of this channel in one query.

Reply:
Also we probably want to fetch the last message from the channel dynamically, instead of relying on `ref`.

Reply:
@sea-odoo . you mean we have to mark as read whole "general" channel? as there are total 5 messages in the channel.
And as per the task need to mark as read first message of the "general" channel.

Reply:
@seb-odoo  Wrong tag, this one is for you 
f.y.i : @dhp-odoo

Reply:
The goal is to remove the notification and the counter after a fresh install, so all messages present in the channel have to be marked as read for all users.

Reply:
@seb-odoo just have a look.